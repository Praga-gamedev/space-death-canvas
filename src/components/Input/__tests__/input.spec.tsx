import React from 'react';
import renderer from 'react-test-renderer';

import { Input } from '../index';
import { mount } from 'enzyme';

describe('Input component', () => {
    test('input rendered correctly', () => {
        const InputComponent = renderer.create(<Input />).toJSON();
        expect(InputComponent).toMatchSnapshot();
    });

    test('label not exist when label prop undefined', () => {
        const InputComponent = mount(<Input />);
        const InputLabel = InputComponent.find({
            'data-test-id': 'input-label',
        }).hostNodes();
        expect(InputLabel).toHaveLength(0);
    });

    test('label exist when label prop defined', () => {
        const labelText = 'text';
        const InputComponent = mount(<Input label={labelText} />);
        const InputLabel = InputComponent.find({
            'data-test-id': 'input-label',
        }).hostNodes();

        expect(InputLabel).toHaveLength(1);
        expect(InputLabel.at(0).text()).toBe(labelText);
    });

    test('change event working', () => {
        const onChange = jest.fn();
        const props = { onChange };
        const InputComponent = mount(<Input {...props} />);

        InputComponent.find('input').simulate('change');

        expect(onChange).toBeCalledTimes(1);
    });
});
