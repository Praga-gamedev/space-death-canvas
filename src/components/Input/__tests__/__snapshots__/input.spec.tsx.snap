// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component input rendered correctly 1`] = `
<div
  className="css-1vr8bhw"
>
  <input
    className="css-1k2zxyw"
  />
</div>
`;
